### PR TITLE
feat: only require screen capture and accessibility permissions

### PR DESCRIPTION
**Pull request by r17x**
## Changes

* Removed Audio (Microphone) Request permission on first-run.
* Request permission when user click Audio (Microphone) Selection - (only show when not `granted`).

### Attachments

| Granted | Not Granted |
|--------|--------|
| image | image |
|  image | |

/claim #91

Ready for review @richiemcilroy

### CODE REVIEW

**Comment by richiemcilroy** [comment]
Hey @r17x this is awesome! Thanks for submitting so quickly. 

Would you be able to also do this for webcam as well?

Make it so that camera permission is not required at startup, until you decide to (by clicking the list). 

Also, willing to add another $100 to the bounty if you can get the green MacBook webcam light to stop showing when no webcam is currently selected.

**Comment by r17x** [comment]
Already updated the PR @richiemcilroy 

Here's videos for another $100 bounty.

https://github.com/user-attachments/assets/28500477-9a33-43ea-bdc3-8ca8883d7d74

**Comment by r17x** [comment]
Heres for camera (same behavior with `microphone`). @richiemcilroy 

https://github.com/user-attachments/assets/1dfdf7b8-599b-4915-bc5e-7aa11f0e041e

**Comment by richiemcilroy** [comment]
Hey @r17x! looks awesome :)

One problem I've noticed is that once you do give permission, you only see the devices (webcam etc) once you right click and refresh.

Is there a way to make it so that we check again/refresh the list once the permission has been granted?

Also if it's possible to not open the dropdown list when we're actually requesting the permission.

So technically the entire "No Camera" button should request for permission on click, instead of just "Request Permission".

**Actions:**
	1.	Fix Device List Update: Automatically refresh the list of devices (webcam, etc.) once permission is granted, without requiring a manual right-click or refresh.
	2.	Prevent Dropdown from Opening During Permission Request: Ensure that the dropdown list for devices doesn’t open while the app is requesting camera permissions.
	3.	Update “No Camera” Button Behavior: Modify the “No Camera” button to request permission when clicked, instead of only the “Request Permission” button initiating the permission request.

File: apps/desktop/src/routes/(window-chrome)/index.tsx [hunk 1]
```diff
@@ -292,12 +301,21 @@ export default function () {
...
-            commands.setRecordingOptions({
-              ...options.data,
-              cameraLabel: d.deviceId ? d.label : null,
-            });
+          onChange={(device: MediaDeviceInfo | null) => {
+            if (options.data?.captureTarget) {
+              if (device) {
+                commands.setRecordingOptions({
+                  ...options.data,
+                  cameraLabel: device && device?.deviceId ? device?.label : null,
```
Comment (galabra) [inline apps/desktop/src/routes/(window-chrome)/index.tsx]: `device && device?.deviceId` is equivalent to `device?.deviceId`

**Comment by richiemcilroy** [comment]
Let me know your thoughts @r17x whenever you get a chance - would love to get this live ASAP

**Comment by r17x** [comment]
@richiemcilroy Here's my new updates:

> 1. Fix Device List Update: Automatically refresh the list of devices (webcam, etc.) once permission is granted, without requiring a manual right-click or refresh.

Automatically update the list options devices (you must be click the options lists for showing the list of devices). Anyway, I didn't understood what's you mean `right-click` or `refresh`. 

> 2. Prevent Dropdown from Opening During Permission Request: Ensure that the dropdown list for devices doesn’t open while the app is requesting camera permissions.

Fixed

> 3. Update “No Camera” Button Behavior: Modify the “No Camera” button to request permission when clicked, instead of only the “Request Permission” button initiating the permission request.

Fixed


https://github.com/user-attachments/assets/fbac08c3-638f-4a08-ba9c-8e57eb7e9276

File: apps/desktop/src/utils/queries.ts [hunk 2]
```diff
@@ -51,13 +61,42 @@ export function createWindowsQuery() {
 
 export function createAudioDevicesQuery() {
   const devices = createQuery(() => listAudioDevices);
-  createTimer(() => devices.refetch(), 1000, setInterval);
```
Comment (r17x) [inline apps/desktop/src/utils/queries.ts:54]: I removed this and set using `queryOptions -> refetchInterval: 1000` - same way

File: apps/desktop/src/routes/(window-chrome)/index.tsx (hunk 1 above)
Comment (r17x) [reply apps/desktop/src/routes/(window-chrome)/index.tsx]: Nice catch, thank you @galabra

**Pull request merged**